### ISSUE TITLE
Nested dependency doesn't update

### ISSUE THREAD

**Issue by Bhirahaspathi-Sairam** [rjsf-team/react-jsonschema-form#3894]
### Prerequisites

- [X] I have read the [documentation](https://rjsf-team.github.io/react-jsonschema-form/docs)

### What theme are you using?

core

### What is your question?

I have a react JSON schema like this 

`{
  "type": "object",
  "properties": {
    "platform": {
      "title": "",
      "$ref": "#/definitions/platform"
    }
  },
  "definitions": {
    "username": {
      "type": "string",
      "title": "Enter Username"
    },
    "url": {
      "type": "string",
      "title": "Enter url"
    },
    "lineId": {
      "type": "string",
      "title": "Enter LineID"
    },
    "platform": {
      "type": "object",
      "properties": {
        "platform": {
          "type": "string",
          "title": "Select Platform",
          "enum": [
            "--select--",
            "Facebook",
            "Instagram",
            "Line"
          ],
          "default": "--select--"
        }
      },
      "dependencies": {
        "action": {
          "oneOf": [
            {
              "properties": {
                "action": {
                  "const": "Follow"
                },
                "username": {
                  "$ref": "#/definitions/username"
                }
              }
            },
            {
              "properties": {
                "action": {
                  "const": "Share"
                },
                "url": {
                  "$ref": "#/definitions/url"
                }
              }
            },
            {
              "properties": {
                "action": {
                  "const": "Add"
                },
                "lineId": {
                  "$ref": "#/definitions/lineId"
                }
              }
            }
          ]
        },
        "platform": {
          "oneOf": [
            {
              "properties": {
                "platform": {
                  "enum": [
                    "Facebook"
                  ]
                },
                "action": {
                  "type": "string",
                  "title": "Select Action",
                  "enum": [
                    "Follow",
                    "Share"
                  ],
                  "enumNames": [
                    "Follow",
                    "Share"
                  ]
                }
              }
            },
            {
              "properties": {
                "platform": {
                  "enum": [
                    "Instagram"
                  ]
                },
                "action": {
                  "type": "string",
                  "title": "Select Action",
                  "enum": [
                    "Follow"
                  ],
                  "enumNames": [
                    "Follow"
                  ]
                }
              }
            },
            {
              "properties": {
                "platform": {
                  "enum": [
                    "Line"
                  ]
                },
                "action": {
                  "type": "string",
                  "title": "Select Action",
                  "enum": [
                    "Add"
                  ],
                  "enumNames": [
                    "Add"
                  ]
                }
              }
            }
          ]
        }
      }
    }
  }
}`

on selecting platform an action field is rendered and based on the action field i am rendering different text fields. After Selecting a platform and action the third text field is displayed, However when i change the platform again the text field is still rendered on the screen .

Is it possible to hide the third field when i re select the platform ?

**Comment by heath-freenome** [comment]
Here is a link to the [playground](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJwbGF0Zm9ybSI6eyJwbGF0Zm9ybSI6IkZhY2Vib29rIn19LCJzY2hlbWEiOnsidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsicGxhdGZvcm0iOnsidGl0bGUiOiIiLCIkcmVmIjoiIy9kZWZpbml0aW9ucy9wbGF0Zm9ybSJ9fSwiZGVmaW5pdGlvbnMiOnsidXNlcm5hbWUiOnsidHlwZSI6InN0cmluZyIsInRpdGxlIjoiRW50ZXIgVXNlcm5hbWUifSwidXJsIjp7InR5cGUiOiJzdHJpbmciLCJ0aXRsZSI6IkVudGVyIHVybCJ9LCJsaW5lSWQiOnsidHlwZSI6InN0cmluZyIsInRpdGxlIjoiRW50ZXIgTGluZUlEIn0sInBsYXRmb3JtIjp7InR5cGUiOiJvYmplY3QiLCJwcm9wZXJ0aWVzIjp7InBsYXRmb3JtIjp7InR5cGUiOiJzdHJpbmciLCJ0aXRsZSI6IlNlbGVjdCBQbGF0Zm9ybSIsImVudW0iOlsiLS1zZWxlY3QtLSIsIkZhY2Vib29rIiwiSW5zdGFncmFtIiwiTGluZSJdLCJkZWZhdWx0IjoiLS1zZWxlY3QtLSJ9fSwiZGVwZW5kZW5jaWVzIjp7ImFjdGlvbiI6eyJvbmVPZiI6W3sicHJvcGVydGllcyI6eyJhY3Rpb24iOnsiY29uc3QiOiJGb2xsb3cifSwidXNlcm5hbWUiOnsiJHJlZiI6IiMvZGVmaW5pdGlvbnMvdXNlcm5hbWUifX19LHsicHJvcGVydGllcyI6eyJhY3Rpb24iOnsiY29uc3QiOiJTaGFyZSJ9LCJ1cmwiOnsiJHJlZiI6IiMvZGVmaW5pdGlvbnMvdXJsIn19fSx7InByb3BlcnRpZXMiOnsiYWN0aW9uIjp7ImNvbnN0IjoiQWRkIn0sImxpbmVJZCI6eyIkcmVmIjoiIy9kZWZpbml0aW9ucy9saW5lSWQifX19XX0sInBsYXRmb3JtIjp7Im9uZU9mIjpbeyJwcm9wZXJ0aWVzIjp7InBsYXRmb3JtIjp7ImVudW0iOlsiRmFjZWJvb2siXX0sImFjdGlvbiI6eyJ0eXBlIjoic3RyaW5nIiwidGl0bGUiOiJTZWxlY3QgQWN0aW9uIiwiZW51bSI6WyJGb2xsb3ciLCJTaGFyZSJdLCJlbnVtTmFtZXMiOlsiRm9sbG93IiwiU2hhcmUiXX19fSx7InByb3BlcnRpZXMiOnsicGxhdGZvcm0iOnsiZW51bSI6WyJJbnN0YWdyYW0iXX0sImFjdGlvbiI6eyJ0eXBlIjoic3RyaW5nIiwidGl0bGUiOiJTZWxlY3QgQWN0aW9uIiwiZW51bSI6WyJGb2xsb3ciXSwiZW51bU5hbWVzIjpbIkZvbGxvdyJdfX19LHsicHJvcGVydGllcyI6eyJwbGF0Zm9ybSI6eyJlbnVtIjpbIkxpbmUiXX0sImFjdGlvbiI6eyJ0eXBlIjoic3RyaW5nIiwidGl0bGUiOiJTZWxlY3QgQWN0aW9uIiwiZW51bSI6WyJBZGQiXSwiZW51bU5hbWVzIjpbIkFkZCJdfX19XX19fX19LCJ1aVNjaGVtYSI6e30sInRoZW1lIjoibWF0ZXJpYWwtdWktNSIsImxpdmVTZXR0aW5ncyI6e319) with your example. I'm noticing that when you select a different platform, the `action` data remains, which is why the input field shows up.

**Comment by Bhirahaspathi-Sairam** [comment]
yes the action field remains.Even though action field is being changed in the ui i.e dropdown , the formdata retains the old action value. Any idea how this can be resolved?

**Comment by heath-freenome** [comment]
@Bhirahaspathi-Sairam See my updated comment above

**Comment by Bhirahaspathi-Sairam** [comment]
@heath-freenome is this issue will be resolved in the future ?

**Comment by magaton** [comment]
I just hit the same problem.
Basically, if the value of the conditional field is changed, every field that was rendered based on that property should be reset, before the condition is evaluated (based on the new selected option).

Is this statement correct? If so, is it something that could be considered as a feature?

I understand that with `onChange`  everything can be manipulated, but this requires hardcoding of field names and custom JS code that pretty much defies the purpose of this library.

**Comment by magaton** [comment]
Hello, just to bounce this issue, as it is still unanswered.

**Comment by heath-freenome** [comment]
@magaton Yes, this is a new feature request. Thanks for following up. We'd love help building it.
